internal/store/job.go

```diff
@@ -163,16 +164,49 @@ var (
 		),
 		*generator.NewFamilyGenerator(
 			"kube_job_status_failed",
-			"The number of pods which reached Phase Failed.",
+			"The number of pods which reached Phase Failed & their failure reason.",
 			metric.Gauge,
 			"",
 			wrapJobFunc(func(j *v1batch.Job) *metric.Family {
-				return &metric.Family{
-					Metrics: []*metric.Metric{
-						{
-							Value: float64(j.Status.Failed),
+				var ms []*metric.Metric
+
+				if float64(j.Status.Failed) == 0 {
+					return &metric.Family{
+						Metrics: []*metric.Metric{
+							{
+								Value: float64(j.Status.Failed),
+							},
 						},
-					},
+					}
+				}
+
+				for _, condition := range j.Status.Conditions {
+
+					if condition.Type == v1batch.JobFailed {
+						reasonKnown := false
+						for _, reason := range jobFailureReasons {
+							reasonKnown = reasonKnown || failureReason(&condition, reason)
+
+							// for known reasons
+							ms = append(ms, &metric.Metric{
+								LabelKeys:   []string{"condition", "reason"},
+								LabelValues: []string{string(condition.Type), reason},
```

Comment:
The condition will always be `Failed` here, is that what you want?

Shouldn't this label contain the value of `condition.Status` instead? If so, you might want to have a look at the `addConditionMetrics` function.

Reply:
Yes, we want only the `Failed` conditions as this metric is `kube_job_status_failed`.
If required then we can change the metric to `kube_job_status` and include other statuses as well.

Reply:
Then, I don't think we need to add this dimension to the metric. We already have the information that the job failed in the metric name, it would be redundant to also have it in a label.